src/widget/wtracktableview.cpp

```diff
@@ -1074,6 +1087,49 @@ void WTrackTableView::contextMenuEvent(QContextMenuEvent* event) {
             }
         }
         m_pMenu->addMenu(m_pBPMMenu);
+
+        // Track color menu only appears if at least one track is selected
+        if (indices.size()) {
+            // Get color of first selected track
+            int column = trackModel->fieldIndex(LIBRARYTABLE_COLOR);
+            QModelIndex index = indices.at(0).sibling(indices.at(0).row(), column);
+            mixxx::RgbColor::optional_t trackColor = std::nullopt;
+            if (!index.data().isNull()) {
+                trackColor = mixxx::RgbColor::optional(index.data().toUInt());
+            }
+
+            // Check if all other selected tracks have the same color
+            for (int i = 1; i < indices.size(); ++i) {
+                int row = indices.at(i).row();
+                QModelIndex index = indices.at(i).sibling(row, column);
+
+                mixxx::RgbColor::optional_t otherTrackColor = std::nullopt;
+                if (!index.data().isNull()) {
+                    otherTrackColor = mixxx::RgbColor::optional(index.data().toUInt());
+                }
+
+                if (trackColor != otherTrackColor) {
+                    trackColor.reset();
```

Comment:
```
trackColor = std::nullopt;
```

Reply:
Can you explain why? I'll change this but I don't get how this is better than reset().

Reply:
Support for std::optional is **very** limited in the old Xenial GCC, reset() doesn't seem to be supported. That's why the Travis build fails.

Reply:
Ah ok ;-)